tests/controller_tests/test_ingest.py

```diff
@@ -138,3 +142,21 @@ def test_extract_data_extracts_from_multiple_sheets(example_data_model_file):
     assert len(workbook) > 1
     assert isinstance(workbook, dict)
     assert isinstance(list(workbook.values())[0], pd.DataFrame)
+
+
+def test_save_submission_file(app_ctx):
+    sub = Submission(
+        submission_id="1",
+        reporting_period_start=datetime.now(),
+        reporting_period_end=datetime.now(),
+        reporting_round=1,
+    )
+    db.session.add(sub)
+
+    filename = "example.xlsx"
+    filebytes = b"example file contents"
+    file = FileStorage(BytesIO(filebytes), filename=filename)
+
+    save_submission_file(file, submission_id=sub.submission_id)
+    assert Submission.query.first().submission_filename == filename
+    assert Submission.query.first().submission_file == filebytes
```

Comment:
Can we persist this as a dataframe or some picked python format in a database Text field or similar? Not sure if it's a great idea to store raw files directly in the db?

Reply:
A potential problem with that approach is that if we pickle the DataFrame and the dependencies go out of sync we could run into problems unpickling later on. I have experienced this problem with pickling before..

Reply:
OK, let's leave as binary for now, but add a TODO to migrate to S3.

Reply:
Added a TODO